src/main/java/org/springframework/data/elasticsearch/core/MappingBuilder.java

```diff
@@ -251,6 +255,9 @@ private static void addSingleFieldMapping(XContentBuilder xContentBuilder, java.
 		if (isNotBlank(fieldAnnotation.analyzer())) {
 			xContentBuilder.field(FIELD_INDEX_ANALYZER, fieldAnnotation.analyzer());
 		}
+		if (fieldAnnotation.termVector() != FIELD_TERM_VECTOR_DEFAULT) {
+			xContentBuilder.field(FIELD_TERM_VECTOR, fieldAnnotation.termVector());
+		}

```

Comment:
To avoid doing it twice, please move this into the method `MappingBuilder#addFieldMappingParameters()`